evm/src/witness/util.rs

```diff
@@ -200,6 +195,26 @@ pub(crate) fn mem_write_gp_log_and_fill<F: Field>(
     op
 }
 
+pub(crate) fn mem_write_partial_log_and_fill<F: Field>(
+    address: MemoryAddress,
+    state: &GenerationState<F>,
+    row: &mut CpuColumnsView<F>,
+    val: U256,
+) -> MemoryOp {
+    let op = mem_write_log(MemoryChannel::PartialChannel, address, state, val);
+
+    let channel = &mut row.partial_channel;
+    assert!(channel.used.is_zero());
+    // channel.used = F::ONE;
```

Comment:
Maybe you should uncomment this, since I believe you're setting it to one every time you actually call this function (and then you could remove ` row.partial_channel.used = F::ONE;` in the statement `if let Some(log) = write`, in `push_with_write`).